persistence/src/main/java/org/cbioportal/genome_nexus/persistence/internal/EnsemblRepositoryImpl.java

```diff
@@ -20,6 +25,53 @@ public EnsemblRepositoryImpl(MongoTemplate mongoTemplate)
 
     public static final String CANONICAL_TRANSCRIPTS_COLLECTION = "ensembl.canonical_transcript_per_hgnc";
     public static final String TRANSCRIPTS_COLLECTION = "ensembl.biomart_transcripts";
+    public static final String HGNC_COLLECTION = "hgnc";
+
+    @Override
+    public EnsemblGene getCanonicalEnsemblGeneIdByHugoSymbol(String hugoSymbol) {
+        BasicDBList or = new BasicDBList();
+
+        BasicDBObject approvedSymbolQuery = new BasicDBObject();
+        // case insensitive exact match query
+        approvedSymbolQuery.put("approved_symbol",
+            new BasicDBObject("$regex", "^" + hugoSymbol + "$")
+                .append("$options", "i"));
+        or.add(approvedSymbolQuery);
+
+        // synonyms
+        BasicDBObject synonymQuery = new BasicDBObject();
+        synonymQuery.put("synonyms",
+            new BasicDBObject("$regex", "(^| |,)" + hugoSymbol + "($| |,)")
+                .append("$options", "i"));
+        or.add(synonymQuery);
+
+        // prev symbols
+        BasicDBObject prevSymbolsQuery = new BasicDBObject();
+        prevSymbolsQuery.put("previous_symbols",
+            new BasicDBObject("$regex", "(^| |,)" + hugoSymbol + "($| |,)")
+                .append("$options", "i"));
+        or.add(prevSymbolsQuery);
+
+        Cursor canonicalCursor = mongoTemplate.getCollection(HGNC_COLLECTION).find(new BasicDBObject("$or", or));
+        if (canonicalCursor.hasNext()) {
+            BasicDBObject canonicalGeneId = (BasicDBObject) canonicalCursor.next();
```

Comment:
Instead of using a `BasicDBObject` instance, maybe map `HGNC_COLLECTION` to a proper Java model? Then `(String) canonicalGeneId.get("ensembl_id")` would become `canonicalGeneId.getEnsemblId()`. Then we can also do something like this (assuming that we are also using the `Query` instance):
```Java
List<CanonicalGeneId> canonicalGeneIds = this.mongoTemplate.find(query, CanonicalGeneId.class, HGNC_COLLECTION);
```